router/router.go

```diff
@@ -1594,8 +1589,11 @@ func (rt *HandleT) commitStatusList(responseList *[]jobResponseT) {
 				}
 
 				rt.failureMetricLock.Lock()
-				failureMetricVal := failureMetric{RouterDestination: rt.destName, UserId: resp.userID, RouterAttemptNum: resp.status.AttemptNum, ErrorCode: resp.status.ErrorCode, ErrorResponse: resp.status.ErrorResponse}
-				rt.failuresMetric[event] = append(rt.failuresMetric[event], failureMetricVal)
+				_, ok := rt.failuresMetric[event][string(resp.status.ErrorResponse)]
```

Comment:
truncating the `ErrorResponse` to a specific limit, e.g. 1000 characters would be useful as well. We don't care about it being a parseable json, do we?

Reply:
I even thought of doing the same but since these end up in snowflake I just let the response be a whole JSON rather than a truncated object in case they need to be queried